syserol/figures/common.py

```diff
@@ -269,26 +269,82 @@ def buildFigure3(legends=True, heatmap=False):
         for aa in ax:
             aa.axis('equal')
     else:
-        ax, f = getSetup((8, 8), (1, 4))
-        
-        cbar_kws = {"orientation":"horizontal"}
+        f = plt.figure(figsize=(10, 15))
+        gs = gridspec.GridSpec(2, 4,
+                            width_ratios=[2, 25, 12, 25],
+                            height_ratios=[1, 1],
+                            wspace=0, hspace=.3
+                            )
+        ax1 = plt.subplot(gs[0])
+        ax2 = plt.subplot(gs[1])
+        ax4 = plt.subplot(gs[3])
+        ax6 = plt.subplot(gs[5])
+        ax8 = plt.subplot(gs[7])
+
+
+        colors = ["blue", "orange", "green", "red"]
+        cmap = sns.color_palette(colors)
+        #cbar_kws = {"location":"top"}
         subs = pd.DataFrame(subjects, columns = [f"Component {i}" for i in np.arange(1, subjects.shape[1] + 1)], index=subjinfo["class.etuv"])
-        plt = sns.heatmap(subs, cmap="PRGn", center=0, xticklabels=True, yticklabels=50, cbar_kws=cbar_kws, ax=ax[0])
-        plt.set_ylabel("")
+        sns.heatmap(subs, cmap="PRGn", center=0, xticklabels=True, yticklabels=False, cbar_kws=dict(fraction=.05, pad=.03), ax=ax2)
 
         rec = pd.DataFrame(receptors, columns = [f"Component {i}" for i in np.arange(1, subjects.shape[1] + 1)], index=detections)
-        sns.heatmap(rec, cmap="PRGn", center=0, yticklabels=True, cbar_kws=cbar_kws, ax=ax[1])
+        sns.heatmap(rec, cmap="PRGn", center=0, yticklabels=True, cbar_kws=dict(fraction=.05, pad=.03), ax=ax4)
 
         ant = pd.DataFrame(antigens, columns=[f"Component {i}" for i in np.arange(1, subjects.shape[1] + 1)], index=antigen)
-        sns.heatmap(ant, cmap="PRGn", center=0, yticklabels=True, cbar_kws=cbar_kws, ax=ax[2])
+        sns.heatmap(ant, cmap="PRGn", center=0, yticklabels=True, cbar_kws=dict(fraction=.05, pad=.03), ax=ax6)
```

Comment:
The color scales are all pretty similar. Perhaps we can just use one color scale to make this a bit cleaner?

Reply:
Do you mean the color bar/legend? If I change this to a wide format, where would you like it? At the end of all 4?

Reply:
Yep. Probably would want it with the first panel.

Reply:
Sorry, do you mean after the first graph? So it would go subjects plot, then color bar, then the other 3 plots?

Reply:
Yep.